avocado/core/requirements/resolver.py

```diff
@@ -0,0 +1,106 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: Red Hat Inc. 2020
+# Authors: Willian Rampazzo <willianr@redhat.com>
+
+"""
+Requirements Resolver module.
+"""
+
+from ..nrunner import RUNNERS_REGISTRY_PYTHON_CLASS, Runnable, Task
+
+SUPPORTED_REQUIREMENTS = {}
+
+
+def create_requirements_tasks(tasks):
+    """Entry point for the Requirements Resolver.
+
+    This function isolates the Requirements Resolver object into this module.
+
+    :param tasks: List of tasks
+    :type tasks: list
+    """
+    resolver = RequirementsResolver(tasks)
+    return resolver.create_requirements_tasks()
+
+
+class UnsupportedRequirementType(Exception):
+    pass
+
+
+class RequirementsResolver:
+    """Requirements Resolver class."""
+
+    def __init__(self, tasks):
+        """
+        :param tasks: List of tasks to be resolved
+        :type tasks: list
+        """
+        self._tasks = tasks
+
+    def create_requirements_tasks(self):
+        """Create the list of tasks based on the test requirements."""
+        requirements_tasks = []
+        for task in self._tasks:
+            if not task.runnable.requirements:
+                continue
+
+            uris = [status_service.uri
+                    for status_service in task.status_services]
+
+            for requirement in task.runnable.requirements:
+                try:
+                    type_klass = SUPPORTED_REQUIREMENTS[requirement['type']]
```

Comment:
No way this could just be `type_class` which is also not a reserved word and easier to read?

Reply:
No problem on my side. I just noticed it is a common practice in many projects to use this kind of naming, and as we are already using it on our code, I thought it would not hurt.

Reply:
Yes, I understand and I have seen this before in the current code base too so there is nothing wrong with your line whatsoever. However, nothing stops us from making a step against it now and completing it in the future and avoid creating raising extra questions to newcomer readers of the code.

Reply:
But as you wish, this is a minor thing that just sticks out a bit.